hack/imagebuilder/main.go

```diff
@@ -0,0 +1,296 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"sync"
+
+	"context"
+
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	defaultArch = "linux/amd64"
+	allArch     = "all"
+
+	gatherStaicScriptName = "gather-static.sh"
+
+	// Relative to root of the repo
+	defaultProwImageListFile = "prow/.prow-images.yaml"
+
+	defaultWorkersCount = 10
+
+	// noOpKoDocerRepo is used when images are not pushed
+	noOpKoDocerRepo = "do.not/matter/at/all"
+)
+
+var (
+	rootDir     string
+	otherArches = []string{
+		"arm64",
+		"s390x",
+		"ppc64le",
+	}
+	defaultTags = []string{
+		"latest",
+		"latest-root",
+	}
+)
+
+func init() {
+	out, err := runCmd("git", "rev-parse", "--show-toplevel")
+	if err != nil {
+		logrus.WithError(err).Error("Failed getting git root dir")
+		os.Exit(1)
+	}
+	rootDir = out
+}
+
+func init() {
+	_, err := runCmdInDir(path.Join(rootDir, "hack/tools"), "go", "build", "-o", path.Join(rootDir, "_bin/ko"), "github.com/google/ko")
+	if err != nil {
+		logrus.WithError(err).Error("Failed ensure ko")
+		os.Exit(1)
+	}
+}
+
+type options struct {
+	koDockerRepo      string
+	prowImageListFile string
+	workers           int
+	push              bool
+}
+
+func runCmdInDir(dir, cmd string, args ...string) (string, error) {
+	command := exec.Command(cmd, args...)
+	if dir != "" {
+		command.Dir = dir
+	}
+	stdOut, err := command.StdoutPipe()
+	if err != nil {
+		return "", err
+	}
+	stdErr, err := command.StderrPipe()
+	if err != nil {
+		return "", err
+	}
+	if err := command.Start(); err != nil {
+		return "", err
+	}
+	scanner := bufio.NewScanner(stdOut)
+	var allOut string
+	for scanner.Scan() {
+		out := scanner.Text()
+		allOut = allOut + out
+		logrus.WithField("cmd", command.Args).Info(out)
+	}
```

Comment:
Why log every line of output separately rather than using `io.ReadAll` like you do for stderr?

Reply:
typescript rollup takes quite some time, I think it is better to stream logs so we know it's still ongoing